pkg/config/config.go

```diff
@@ -667,15 +661,18 @@ func LoadWithDefaults(getCfgPaths PathsGetter) (*Config, LoadWithDefaultsDetails
 	}, nil
 }
 
-// FromEnvOrFlag resolves and returns paths for config files.
+// FromProvider resolves and returns paths for config files.
 // It reads them the 'BOTKUBE_CONFIG_PATHS' env variable. If not found, then it uses '--config' flag.
-func FromEnvOrFlag() []string {
-	envCfgs := os.Getenv("BOTKUBE_CONFIG_PATHS")
-	if envCfgs != "" {
-		return strings.Split(envCfgs, ",")
+func FromProvider(gql *config.GqlClient) (config.YAMLFiles, error) {
+	var provider config.Provider
+	if os.Getenv("CONFIG_SOURCE_IDENTIFIER") != "" {
+		provider = config.NewGqlProvider(*gql)
+	} else if os.Getenv("BOTKUBE_CONFIG_PATHS") != "" {
+		provider = config.NewEnvProvider()
+	} else {
+		provider = config.NewFileSystemProvider(configPathsFlag)
 	}
-
-	return configPathsFlag
+	return provider.Configs(context.Background())
```

Comment:
Shouldn't context come from the calling function?

Reply:
calling function does not have context, and I needed to pass context since there is a such contract

Reply:
Can we take this context? https://github.com/kubeshop/botkube/blob/3aa76fade5ba174b6829942a0bc095d46eb21558/cmd/botkube/main.go#L105

Reply:
But it is set after config load operation.Even analytics initialization is done without context.
![Screen Shot 2022-12-23 at 15 01 19](https://user-images.githubusercontent.com/1237982/209333096-221280d0-d50a-452f-8b62-a20e37273537.png)

Reply:
Just a suggestion. The ctx could me moved.

Reply:
I created a follow up task for that https://github.com/kubeshop/botkube/issues/908